# HG changeset patch
# User sbr <sebastien@brissaud.name>
# Date 1375248444 -7200
#      Wed Jul 31 07:27:24 2013 +0200
# Node ID b26cab0455824c202c1091ae9e4fa41419db9471
# Parent  c747935615b5daebd80391cf2c1618c7f6bf9b31
-Fix (r21448): The base horizontal offset of NWidgetMatrix children should also be right-shifted if there is no scrollbar when drawing in RTL mode.

diff --git a/src/widget.cpp b/src/widget.cpp
--- a/src/widget.cpp
+++ b/src/widget.cpp
@@ -1674,6 +1674,9 @@ void NWidgetMatrix::GetScrollOffsets(int
 				base_offs_x = -this->sb->GetPosition() + start_x * this->widget_w;
 			}
 		}
+	} else {
+		/* Also right-shift the base horizontal offset when drawing in RTL mode if there is no scrollbar. */
+		if (_current_text_dir == TD_RTL) base_offs_x = this->pip_pre + this->widget_w * (this->widgets_x - 1) - this->pip_post;
 	}
 }
 
